pkg/controller/operators/catalog/operator.go

```diff
@@ -727,10 +727,19 @@ func (o *Operator) syncRegistryServer(logger *logrus.Entry, in *v1alpha1.Catalog
 
 	// requeue the catalog sync based on the polling interval, for accurate syncs of catalogs with polling enabled
 	if out.Spec.UpdateStrategy != nil {
-		logger.Debugf("requeuing registry server sync based on polling interval %s", out.Spec.UpdateStrategy.Interval.Duration.String())
-		resyncPeriod := reconciler.SyncRegistryUpdateInterval(out, time.Now())
-		o.catsrcQueueSet.RequeueAfter(out.GetNamespace(), out.GetName(), queueinformer.ResyncWithJitter(resyncPeriod, 0.1)())
-		return
+		if out.Spec.UpdateStrategy.RegistryPoll != nil {
+			if out.Spec.UpdateStrategy.RegistryPoll.ParsingError != "" {
+				out.SetError(v1alpha1.CatalogSourceIntervalInvalidError, fmt.Errorf(out.Spec.UpdateStrategy.RegistryPoll.ParsingError))
+				if _, err := o.client.OperatorsV1alpha1().CatalogSources(out.GetNamespace()).UpdateStatus(context.TODO(), out, metav1.UpdateOptions{}); err != nil {
```

Comment:
is this `UpdateStatus` call required every catalogsource reconciliation loop if the ParsingError is not blank? Maybe we should only do the status update once? Is there any way to check if that status is set before attempting the update call?

Reply:
Maybe someone can clarify, but I'm not sure that we want to always flip the status.reason to `CatalogSourceIntervalInvalidError` from another status in the case where a parsing error was found on the catalog. This would cause us to stomp our other catalog status updates and potentially constantly update the status.

I think what we want is to provide some kind of long-lived indication that the catalog did have a invalid interval at some point (when it was created), which was changed to the default, with a message indicating what happened. WDYT about using a condition on the status instead of updating the status.Reason directly?

Reply:
I see what you're saying, since we are checking the polling error in this final sequence of events, it may not necessarily be overwritten. But if some other condition earlier in the chain is met, we update the status, and then when it's resolved we will update here. This particular status (invalid polling interval) doesn't feel like a level-based error, if that makes sense. I'm still a little worried about the potential for hotlooping

Do we ever clear the `spec.UpdateStrategy.RegistryPoll.ParsingError` field? Because it feels like strange behavior to always update the status to point to an invalid catalog error, even if the polling error has been resolved (by defaulting) and the catalog is no longer invalid.  

Also, I believe we can simply set 
```
syncError = fmt.Errorf(out.Spec.UpdateStrategy.RegistryPoll.ParsingError)
out.SetError(v1alpha1.CatalogSourceIntervalInvalidError, syncError)
return
```
in the code and not have to call `UpdateStatus()` directly, the control loop will update the catalog for us

Reply:
@exdx so the `spec.UpdateStrategy.RegistryPoll.ParsingError` does clear when we unmarshall the incoming object [here](https://github.com/operator-framework/api/blob/master/pkg/operators/v1alpha1/catalogsource_types.go#L144-L148). I added a test to verify that https://github.com/operator-framework/operator-lifecycle-manager/pull/2447/files#diff-65a3cf2eeb27e70487e6159bd540e1d057e3bc79bea13e4c874a147d6f147469R1195

Reply:
I'm guessing if this test passes that'll take care of the hotlooping concerns too.